hack/lib/util.sh

```diff
@@ -474,9 +474,25 @@ kube::util::go_install_from_commit() {
   local -r pkg=$1
   local -r commit=$2
 
+  # force creating a new temp dir so that the GOPATH is not polluted by previous
+  # calls to go_install_from_commit
+  KUBE_TEMP=""
   kube::util::ensure-temp-dir
   mkdir -p "${KUBE_TEMP}/go/src"
   GOPATH="${KUBE_TEMP}/go" go get -d -u "${pkg}"
+  # If the requested package has vendored dependencies, go get -d won't install
+  # them into GOPATH. This may break things if the dependencies weren't vendored
+  # at the commit requested.
+  # As a hacky workaround, copy the dependencies from tip into GOPATH.
+  (
+    cd "${KUBE_TEMP}/go/src/${pkg}"
+    while [[ "$(pwd)" != "${KUBE_TEMP}/go/src" && "$(pwd)" != "/" ]]; do
```

Comment:
I know temp directories on macOS are weird, so I'd appreciate someone with a Mac testing to make sure this works correctly.

Reply:
will do. also @cblecker

Reply:
held to test https://github.com/kubernetes/kubernetes/pull/64929#event-1671529512 instead, running a few things to sanity check on mac